### ISSUE TITLE
Query about `-a` flag

### ISSUE THREAD

**Issue by dorchard** [camfort/fortran-src#252]
The -a flag let's us pass "lex" or "parse" to tell fortran-src what to do:
https://github.com/camfort/fortran-src/blob/cca921b0038e6df76413a267ed58f39b88ca8917/app/Main.hs#L336

But do we want this to be more general (to allow any action?).
I'm not 100% sure why we have this option anyway, but I wonder why its so specific.

**Comment by raehik** [comment]
You're right -- it would seem sensible if the "lex or parse" choice were used further down the line for other commands, but it's not. It's useful to check the lexer output, so I'd want to keep it. It's exposed differently to the rest of the functionality (which is all option-based). I'm not sure what else we can do other than clarify the option as you did, without altering the interface a fair amount. (Mused on this in #177.)

Based on personal experience, I find programs that expose their functionality through commands and subcommands are more discoverable than fortran-src's option-based approach. What about an interface like

```bash
fortran-src typecheck mod1.f mod2.f
```

or closer to existing syntax

```bash
# --action now takes lex,parse,typecheck,rename,bblocks,...
fortran-src --action typecheck mod1.f mod2.f
```

**Comment by dorchard** [comment]
The latter would be very easy to implement so that might be the best choice.
